citar.el

```diff
@@ -430,14 +430,39 @@ REBUILD-CACHE and FILTER."
            (category . file))
        (complete-with-action action files string predicate)))))
 
-(defun citar-select-group-related-sources (file transform)
-  "Group by FILE by source, TRANSFORM."
-    (let ((extension (file-name-extension file)))
-      (when transform file
-        ;; Transform for grouping and group title display.
+(defun citar-select-resource (files &optional links)
+  "Select resource from a list of FILES, and optionally LINKS."
+  (let* ((files (mapcar
+                 (lambda (cand)
+                   (abbreviate-file-name cand))
+                 files))
+         (resources (append files (remq nil links))))
+    (dolist (item resources)
+      (cond ((string-match "http" item 0)
+             (add-text-properties
+              0 (length item) `(consult-multi (url . ,item)) item))
+            (t
+             (add-text-properties
+              0 (length item) `(consult-multi (file . ,item)) item)))
```

Comment:
Careful, this modifies the strings! I recommend propertize which keeps the original strings intact.

Reply:
Yeah, I had just copied over the code from #491. Fixing this now.